packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/ArrayOfObjectsFunctions.tsx

```diff
@@ -51,64 +41,64 @@ export function ArrayOfObjectsFunctions<
       ? 'inputs.array.action.add-item-select-type'
       : 'inputs.array.action.add-item'
 
+  const insertButtonProps: React.ComponentProps<typeof Button> = {
+    icon: AddIcon,
+    mode: 'ghost',
+    size: 'large',
+    text: t(addItemI18nKey),
+  }
+
+  const insertMenu = useInsertMenuPopover({
+    insertMenuProps: {
+      ...props.schemaType.options?.insertMenu,
+      schemaTypes: props.schemaType.of,
+      onSelect: insertItem,
+    },
+    popoverProps: {
+      placement: 'bottom',
+      fallbackPlacements: ['top'],
+      matchReferenceWidth: props.schemaType.options?.insertMenu?.filter === 'on',
```

Comment:
@mariuslundgard, @rexxars  , I think this logic is a tiny bit flawed now that we have auto-filtering. The original idea was to make the width of the menu match the button if filtering was turned on, but we can't infer from the outside that the menu decides to auto-enable filtering.

But now we might need
1. An alternate way to infer full width
2. A way for the menu to tell the call site that filtering is enabled

What do you think?

---

Array input with no insert menu options configured:
![Screenshot 2024-06-17 at 11 39 19](https://github.com/sanity-io/sanity/assets/4602382/b4e0a782-e658-4b47-8188-082f83b0951d)

Array input with `filter: 'on'`:
![Screenshot 2024-06-17 at 11 44 45](https://github.com/sanity-io/sanity/assets/4602382/5454f9be-378a-401f-8c9d-b92d0631bc68)

Array input with everything configured except `filter`:
![Screenshot 2024-06-17 at 11 42 16](https://github.com/sanity-io/sanity/assets/4602382/d92115b8-959d-42a6-977d-88d5b2befae0)